gen/templates/component/component-name.adb

```diff
@@ -943,12 +943,31 @@ package body Component.{{ name }} is
 
    not overriding procedure Process_Parameter_Update (Self : in out Base_Instance; Par_Update : in out Parameter_Update.T) is
       use Parameter_Operation_Type;
+      use Parameter_Validation_Status;
+      use Parameter_Update_Status;
       Status : Parameter_Update_Status.E := Parameter_Update_Status.Success;
    begin
       case Par_Update.Operation is
          when Stage =>
             -- Stage this parameter.
             Status := Self.Stage_Parameter (Par_Update.Param);
+         when Validate =>
+            -- Pass the staged parameters to the user defined validation function.
+            -- Note that parameter ranges have already been validated as part of staging
+            -- each parameter individually, this function is used for extended,
+            -- user-implemented validation.
+            if Base_Instance'Class (Self).Validate_Parameters_Action (
+{% for par in parameters %}
+               {{ par.name }} => Self.Staged_Parameters.Get_{{ par.name }}{{ "," if not loop.last }}
+{% endfor %}
+            ) /= Valid
+            then
+               -- The default return for this function always returns "Valid", so this
+               -- should be inaccessible.
+               Status := Validation_Error;
+            end if;
+            -- Set status:
+            Status := Validation_Success;
```

Comment:
Don't you need to put this in an `else`?

Reply:
Addressed in [87b53e2](https://github.com/lasp/adamant/pull/64/commits/87b53e25918b9ba58654c3b49c2e6f54074784cb).